### PR TITLE
Update Reference to latest swagger for Runbook Cmdlets

### PR DESCRIPTION
**Pull request by sushil490023**
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

# Changes

With recent changes made to swagger https://github.com/Azure/azure-rest-api-specs/pull/21128/files#diff-8a2476001660339cd88481fb4f06109174b32e1be800923813a9c5ecd0fe53fd

Pattern rule was added to Runbook Name resource to mandatory requirement . However this pattern didnt matched with what is mentioned in public document and whats already being used by customer.

https://learn.microsoft.com/en-us/azure/automation/manage-runbooks

image

This introduced a bug in new  cmdlets generated by usingthis swagger version for few customer.

PR Changes in Swagger (In review ): https://github.com/Azure/azure-rest-api-specs/pull/22352

The changes are used to fix the Name as per doc and fix breaking changes

### CODE REVIEW

File: src/Automation/Automation.Sdk/README.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ input-file:
   - https://github.com/Azure/azure-rest-api-specs/blob/main/specification/automation/resource-manager/Microsoft.Automation/stable/2022-08-08/softwareUpdateConfigurationMachineRun.json
   - https://github.com/Azure/azure-rest-api-specs/blob/main/specification/automation/resource-manager/Microsoft.Automation/stable/2022-08-08/account.json
   - https://github.com/Azure/azure-rest-api-specs/blob/main/specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/webhook.json
-  - https://github.com/Azure/azure-rest-api-specs/blob/main/specification/automation/resource-manager/Microsoft.Automation/stable/2022-08-08/runbook.json
+  - https://github.com/sushil490023/azure-rest-api-specs/blob/feature-FixAutomation/specification/automation/resource-manager/Microsoft.Automation/stable/2022-08-08/runbook.json
```
Comment (isra-fel) [inline src/Automation/Automation.Sdk/README.md:50]:
```suggestion
  - https://github.com/sushil490023/azure-rest-api-specs/blob/feature-FixAutomation/specification/automation/resource-manager/Microsoft.Automation/stable/2022-08-08/runbook.json
```
Hi, we strictly require the swagger to be from the main branch of Azure/azure-rest-api-specs repo. If you are just experimenting or in early dev stage, please make the pull request a draft. Thanks.

Comment (sushil490023) [reply src/Automation/Automation.Sdk/README.md]: Hi @isra-fel I have updated swagger to be from the main branch of Azure/azure-rest-api-specs repo. 
Can you please review it now.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Hi @sushil490023 , please do not modify generated codes directly. please update https://github.com/Azure/azure-powershell/blob/81a90f7cec05f9261974c3dbd50d25ae3d9571f6/src/Automation/Automation.Sdk/README.md?plain=1#L27 to latest swagger you desired and regenerate these codes.

BTW, please check test Commands.Automation.Test.RunbookJobTests.TestImportRunbookPowerShell [FAIL]
    X Commands.Automation.Test.RunbookJobTests.TestImportRunbookPowerShell [242ms]
EXEC : error Message:  [/Users/runner/work/1/s/build.proj]
     System.Management.Automation.ActionPreferenceStopException : The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Unable to find a matching HTTP request for URL 'GET /subscriptions/783fd652-64f3-4680-81e9-0b978c542005/resourceGroups/to-delete-01/providers/Microsoft.Automation/automationAccounts/fbs-aa-01/runbooks/TestRunbook-PowerShellScript?api-version=2022-08-08'. Calling method Item().

**Comment by sushil490023** [comment]
> Hi @sushil490023 , please do not modify generated codes directly. please update
> 
> https://github.com/Azure/azure-powershell/blob/81a90f7cec05f9261974c3dbd50d25ae3d9571f6/src/Automation/Automation.Sdk/README.md?plain=1#L27-L30
> 
> to latest swagger you desired and regenerate these codes.
> BTW, please check test Commands.Automation.Test.RunbookJobTests.TestImportRunbookPowerShell [FAIL] X Commands.Automation.Test.RunbookJobTests.TestImportRunbookPowerShell [242ms] EXEC : error Message: [/Users/runner/work/1/s/build.proj] System.Management.Automation.ActionPreferenceStopException : The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Unable to find a matching HTTP request for URL 'GET /subscriptions/783fd652-64f3-4680-81e9-0b978c542005/resourceGroups/to-delete-01/providers/Microsoft.Automation/automationAccounts/fbs-aa-01/runbooks/TestRunbook-PowerShellScript?api-version=2022-08-08'. Calling method Item().

Hi @BethanyZhou Thanks for reviewing, As changes have been merged to to same API version there has been changes in swagger but not reference hence  generated code did got changed  but not ref in README.
 Merged PR for same : https://github.com/Azure/azure-rest-api-specs/pull/22352

**Comment by BethanyZhou** [comment]
Hi @sushil490023 , thanks for quick response. Code should be changes if new commits are added into main branch no matter API version changes or not. Please have a try.

**Comment by sushil490023** [comment]
> Hi @sushil490023 , thanks for quick response. Code should be changes if new commits are added into main branch no matter API version changes or not. Please have a try.

Sure, Did regeneration again and ran test again for failed test case

**Comment by BethanyZhou** [comment]
Sounds good. Do you plan to release a new version for Az.Automation for your changes? If so, please add change log. For any service, the ChangeLog.md file can be found at src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md

**Comment by sushil490023** [comment]
> Sounds good. Do you plan to release a new version for Az.Automation for your changes? If so, please add change log. For any service, the ChangeLog.md file can be found at src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md

Thanks, Updated ChangeLog.md

**Comment by sushil490023** [comment]
> > Sounds good. Do you plan to release a new version for Az.Automation for your changes? If so, please add change log. For any service, the ChangeLog.md file can be found at src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md
> 
> Thanks, Yes we want new version release Updated ChangeLog.md

File: src/Automation/Automation/ChangeLog.md [hunk 2]
```diff
@@ -19,6 +19,9 @@
 -->
 ## Upcoming Release
 
+## Version 1.9.1
+* Fixed bug: Runbooks Name Pattern failures.
```
[the remark this answers is not in the document]
Comment (BethanyZhou) [reply src/Automation/Automation/ChangeLog.md:23]: Please do not decide version by yourself. Our tools will handle it during the release process.

[the remark this answers is not in the document]
Comment (sushil490023) [reply src/Automation/Automation/ChangeLog.md]: Thanks for correction updated.

**Review by BethanyZhou — APPROVED** [review]

**Comment by sushil490023** [comment]
Thanks @BethanyZhou for approving changes, Can we merge the changes, Also can you let us know what time We can expect new SDK to released for customer.

**Comment by BethanyZhou** [comment]
We will not release new SDK to customer. The generated SDK will work as a private library for our module. You can expert new module on 7th March.

**Pull request merged**